src/dmn-viewer/index.ts

```diff
@@ -0,0 +1,59 @@
+import { css, LitElement, unsafeCSS } from 'lit';
+import { customElement, property } from 'lit/decorators.js';
+import styles from "../shared/styles/reset";
+
+import Viewer from 'dmn-js';
+import ViewerSharedCss from 'dmn-js/dist/assets/dmn-js-shared.css';
+import ViewerDrdCss from 'dmn-js/dist/assets/dmn-js-drd.css';
+import ViewerDecisionTableCss from 'dmn-js/dist/assets/dmn-js-decision-table.css';
+import ViewerLiteralExpressionCss from 'dmn-js/dist/assets/dmn-js-literal-expression.css';
+import ViewerDMNCss from 'dmn-js/dist/assets/dmn-font/css/dmn.css';
+
+@customElement('dmn-viewer')
+export class DMNViewer extends LitElement {
+  private _viewer!: any;
+
+  @property({ attribute: "data-xml" })
+  xml!: string
+
+  constructor() {
+    super();
+  }
+
+  override async firstUpdated() {
+    this._viewer = new Viewer({
+      container: this.renderRoot as HTMLElement
+    });
+
+    try {
+      const { warnings } = await this._viewer.importXML(this.xml);
+
+      if (warnings.length) {
+        console.log('import with warnings', warnings);
+      }
+      else {
+        console.log('import successful');
+      }
+
+      //Why is getActiveViewer necessary?
```

Comment:
ChatGPT says:

In bpmn-js, the method .getActiveViewer() is used to retrieve the currently active viewer or editor instance. This is particularly important in applications where you might have multiple viewers or editors, or when you are dealing with different types of diagram viewers (such as BPMN, DMN, or CMMN).

Here are some specific reasons why you might need to use .getActiveViewer():

Multiple Viewers/Editors: If your application allows users to work with multiple diagrams or diagram types simultaneously, you need to know which viewer or editor instance is currently active in order to interact with it (e.g., to apply updates, access the canvas, or perform other operations).

Context-Sensitive Operations: Some operations may be context-sensitive, meaning they should only be performed on the currently active viewer. For example, zooming, panning, or selecting elements should typically be applied to the diagram the user is currently interacting with.

Plugin Development: When developing plugins or extensions for bpmn-js, you might need to interact with the active viewer instance to extend its functionality, add custom behaviors, or integrate with other parts of your application.

Dynamic UI Updates: If your application's user interface updates dynamically based on the active viewer (e.g., showing properties or controls specific to the current diagram), you need a way to consistently reference the active viewer.

Reply:
I will be asking you in the near future for one of the next assignments to investigate the diagram-js API / architecture so that you can help me to better understand BPMN-JS, DMN-JS and Archimate-JS. What is the same and what is different, how can we add plugins for extra functionality, etc. Not for now but later.